### ISSUE TITLE
flake: `WorkspaceBuildTimings`

### ISSUE THREAD

**Issue by ethanndickson** [coder/internal#200]
Timed out on main `race` today:
```
=== Failed
=== FAIL: coderd  (0.00s)
panic: test timed out after 10m0s
running tests:
	TestWorkspaceBuildTimings/AgentScriptTimings (9m40s)
	TestWorkspaceBuildTimings/EmptyTimings (9m39s)
	TestWorkspaceBuildTimings/ProvisionerTimings (9m39s)
```
https://github.com/coder/coder/actions/runs/11679484254/job/32520700483 (main)
https://github.com/coder/coder/actions/runs/11670350547/job/32494786781 (another branch)

**Comment by BrunoQuaresma** [comment]
I see many other tests are failing too. Sorry for the noob question, but how can I tell if the cause was specifically TestWorkspaceBuildTimings?

**Comment by ethanndickson** [comment]
From what I can tell `TestWorkspaceBuildTimings` is the only test with a known failure reason. When it timed out there were a bunch of goroutines stuck within the test, waiting on the dbmem mutex during `InsertWorkspace`; could potentially be a deadlock.

**Comment by spikecurtis** [comment]
closing in favor of #202 which has a more complete explanation
